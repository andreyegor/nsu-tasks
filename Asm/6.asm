.include "things.asm"
.text
.globl main

main:
    call read_dec
    mv s0 a0
    call div10
    call write_dec
    writi '\n'
    mv a0 s0
    call mod10
    call write_dec
    writi '\n'
    mv a0 s0
    call mul10
    call write_dec
    exit 0

mul10:
    slli t1 a0 1 #multiple 10 TODO
    slli a0 a0 3
    add a0 a0 t1
    ret