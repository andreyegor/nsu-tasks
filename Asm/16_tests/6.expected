ac odio tempor orci dapibus ultrices in iaculis nunc sed augue lacus viverra vitae 
aliquam sem et tortor consequat id porta nibh venenatis cras sed felis eget velit aliquet sagittis id consectetur purus ut faucibus pulvinar elementum integer
arcu vitae elementum curabitur vitae nunc sed velit dignissim sodales ut eu sem integer 
at augue eget arcu dictum varius duis at consectetur lorem donec massa sapien faucibus et molestie ac feugiat sed lectus vestibulum mattis ullamcorper velit sed ullamcorper 
at tellus at urna condimentum mattis pellentesque id nibh tortor id aliquet lectus proin nibh nisl 
condimentum id venenatis a condimentum vitae sapien pellentesque habitant morbi tristique senectus et netus
congue eu consequat ac felis donec et odio pellentesque diam volutpat commodo sed egestas egestas fringilla phasellus faucibus scelerisque eleifend 
consequat mauris nunc congue nisi vitae suscipit tellus mauris a diam maecenas sed enim ut sem viverra aliquet eget
dolor magna eget est lorem ipsum dolor sit amet consectetur adipiscing elit pellentesque habitant morbi tristique
donec pretium vulputate sapien nec sagittis aliquam malesuada bibendum 
eget lorem dolor sed viverra ipsum nunc aliquet bibendum enim facilisis gravida neque convallis a cras semper auctor neque vitae tempus quam pellentesque nec nam
eros in cursus turpis massa tincidunt dui ut ornare lectus sit amet est placerat in egestas erat imperdiet sed
et malesuada fames ac turpis egestas sed tempus urna et pharetra pharetra massa massa ultricies mi quis hendrerit
euismod nisi porta lorem mollis aliquam ut porttitor leo a diam sollicitudin tempor id eu nisl nunc mi ipsum faucibus vitae aliquet nec ullamcorper sit amet risus nullam eget felis eget nunc lobortis
mattis aliquam faucibus purus in massa tempor nec feugiat nisl pretium fusce id velit ut tortor pretium viverra suspendisse potenti nullam ac tortor vitae purus faucibus
mi in nulla posuere sollicitudin aliquam ultrices sagittis orci a scelerisque purus semper eget duis 
morbi tincidunt ornare massa eget egestas purus viverra accumsan in nisl nisi scelerisque eu ultrices vitae auctor eu augue ut lectus arcu bibendum at varius vel pharetra vel turpis nunc
ornare suspendisse sed nisi lacus sed viverra tellus in hac habitasse platea dictumst vestibulum rhoncus est pellentesque elit ullamcorper dignissim cras tincidunt lobortis feugiat vivamus
purus gravida quis blandit turpis cursus in hac habitasse platea dictumst quisque sagittis purus sit amet volutpat
senectus et netus et malesuada fames ac turpis egestas integer eget aliquet nibh praesent tristique magna sit amet
sit amet tellus cras adipiscing enim eu turpis egestas pretium aenean pharetra magna ac placerat vestibulum lectus mauris ultrices
ultricies tristique nulla aliquet enim tortor at auctor urna nunc id cursus metus aliquam eleifend 
vitae justo eget magna fermentum iaculis eu non diam phasellus vestibulum lorem sed risus 